src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -1050,37 +1053,44 @@ static bool isOrContainsErrorType(TypeSymbol typeSymbol)
-                :
-                                                     (Action<DiagnosticBag, MethodSymbol, MethodSymbol, ParameterSymbol, Location>)null,
+                                                 checkReturnType ? reportBadReturn : (ReportMismatchinReturnType<Location>)null,
+                                                 checkParameters ? reportBadParameter : (ReportMismatchInParameterType<Location>)null,
                                                  overridingMemberLocation);
             }
+
+            static void reportBadReturn(DiagnosticBag diagnostics, MethodSymbol overriddenMethod, MethodSymbol overridingMethod, bool blameAttributes, Location location)
+                => diagnostics.Add(blameAttributes ?
+                    ErrorCode.WRN_NullabilityMismatchInReturnTypeOnOverrideBecauseOfAttributes :
+                    ErrorCode.WRN_NullabilityMismatchInReturnTypeOnOverride,
+                    location);
+
+            static void reportBadParameter(DiagnosticBag diagnostics, MethodSymbol overriddenMethod, MethodSymbol overridingMethod, ParameterSymbol overridingParameter, bool blameAttributes, Location location)
+                => diagnostics.Add(
+                    blameAttributes ? ErrorCode.WRN_NullabilityMismatchInParameterTypeOnOverrideBecauseOfAttributes : ErrorCode.WRN_NullabilityMismatchInParameterTypeOnOverride,
+                    location,
+                    new FormattedSymbol(overridingParameter, SymbolDisplayFormat.ShortFormat));
         }
 
         internal static void CheckValidNullableMethodOverride<TArg>(
             CSharpCompilation compilation,
             MethodSymbol overriddenMethod,
             MethodSymbol overridingMethod,
             DiagnosticBag diagnostics,
-            Action<DiagnosticBag, MethodSymbol, MethodSymbol, TArg> reportMismatchInReturnType,
-            Action<DiagnosticBag, MethodSymbol, MethodSymbol, ParameterSymbol, TArg> reportMismatchInParameterType,
+            ReportMismatchinReturnType<TArg> reportMismatchInReturnType,
+            ReportMismatchInParameterType<TArg> reportMismatchInParameterType,
             TArg extraArgument)
         {
             if (!PerformValidNullableOverrideCheck(compilation, overriddenMethod, overridingMethod))
             {
                 return;
             }
 
+            if ((overriddenMethod.FlowAnalysisAnnotations & FlowAnalysisAnnotations.DoesNotReturn) == FlowAnalysisAnnotations.DoesNotReturn &&
+                (overridingMethod.FlowAnalysisAnnotations & FlowAnalysisAnnotations.DoesNotReturn) != FlowAnalysisAnnotations.DoesNotReturn)
+            {
+                diagnostics.Add(ErrorCode.WRN_DoesNotReturnMismatch, overridingMethod.Locations[0], new FormattedSymbol(overridingMethod, SymbolDisplayFormat.MinimallyQualifiedFormat));
+            }
+
             var conversions = compilation.Conversions.WithNullability(true);
             if (reportMismatchInReturnType != null &&
```

Comment:
>reportMismatchInReturnType != null [](start = 16, length = 34)

Please consider checking this condition once, around this case and the one below.